### PR TITLE
SAMA-6 added support for heic image format

### PR DESCRIPTION
**Pull request by Oleksandr1414**
* added support for heic image format
* added a loader when handling images

### CODE REVIEW

File: src/components/screens/chat/chatFormBlocks/ChatFormInputs.js [hunk 1]
```diff
@@ -131,35 +134,61 @@ export default function ChatFormInputs({
...
-          "warning"
-        );
-        return;
-      }
-      if (!file.type.startsWith("image/")) {
-        showCustomAlert("Please select an image file.", "warning");
-        return;
-      }
+    try {
+      for (const file of event.target.files) {
```
Comment (velichko-artem) [inline src/components/screens/chat/chatFormBlocks/ChatFormInputs.js:145]: use forEach array's method instead
const files = event.target.files;
files.forEach(file)

File: src/components/screens/chat/chatFormBlocks/ChatFormInputs.js [hunk 2]
```diff
@@ -131,35 +134,61 @@ export default function ChatFormInputs({
...
+
+          const pngBuffer = await heic2any({
+            blob,
+            toType: "image/png",
+          });
+
+          const pngBlob = new Blob([pngBuffer], { type: "image/png" });
+          const pngFile = new File([pngBlob], file.name, {
+            type: "image/png",
+          });
```
Comment (velichko-artem) [inline src/components/screens/chat/chatFormBlocks/ChatFormInputs.js:169]: I guess, to follow SOLID principles you can export this part of your code to 'utils' and then import here to perform a check:

```
const blob = new Blob([file], { type: "image/heic" });

          const pngBuffer = await heic2any({
            blob,
            toType: "image/png",
          });

          const pngBlob = new Blob([pngBuffer], { type: "image/png" });
          const pngFile = new File([pngBlob], file.name, {
            type: "image/png",
          });
```

Comment (IgorKhomenko) [reply src/components/screens/chat/chatFormBlocks/ChatFormInputs.js:169]: 100% agree

**Review by IgorKhomenko — APPROVED** [review]

File: src/components/screens/chat/chatFormBlocks/ChatFormInputs.js (hunk 1 above)
Comment (Oleksandr1414) [reply src/components/screens/chat/chatFormBlocks/ChatFormInputs.js:145]: But files are Object, so forEach is not the best way to read them, I suggest you use `for`

https://stackoverflow.com/questions/25052758/for-in-vs-object-keys-foreach-without-inherited-properties

**Pull request merged**